src/solvers/refinement/string_refinement.cpp

```diff
@@ -1836,12 +1836,8 @@ static exprt instantiate(
   const exprt &str,
   const exprt &val)
 {
-  const auto indexes = find_indexes(axiom.body, str, axiom.univ_var);
-  INVARIANT(
-    str.id() == ID_array || indexes.size() <= 1,
-    "non constant array should always be accessed at the same index");
   exprt::operandst conjuncts;
-  for(const auto &index : indexes)
+  for(const auto &index : find_indexes(axiom.body, str, axiom.univ_var))
```

Comment:
I'm not sure if I understood this correctly but if there is more than one index via which we access `str`, all of them would get replaced with the same new index `var` which is not correct?

Reply:
If there are several indexes then we get the conjunction of the replacement. 
For instance for a formula `forall q. s[q+x]='a' && s[q]=c` we would get `s[v] = 'a' && s[v-x] = c && s[v+x] = 'a' && s[v] = c`.
I will add this example to the doxygen.